src/agoradatatools/etl/transform/model_details.py

```diff
@@ -0,0 +1,190 @@
+    """
+    # Load all datasets
+    biomarkers_df = datasets["biomarkers"]
+    human_transgene_allele_map_df = datasets["human_transgene_allele_map"]
+    allele_info_df = datasets["allele_info"]
+    model_info_df = datasets["model_info"]
+    pathology_df = datasets["pathology"]
+
+    # Fill all NaN values with empty strings
+    biomarkers_df.fillna("", inplace=True)
+    human_transgene_allele_map_df.fillna("", inplace=True)
+    allele_info_df.fillna("", inplace=True)
+    model_info_df.fillna("", inplace=True)
+    pathology_df.fillna("", inplace=True)
+
+    # Transform text fields to Initial Caps
+    biomarkers_df["sex"] = biomarkers_df["sex"].str.capitalize()
+    biomarkers_df["tissue"] = biomarkers_df["tissue"].apply(
+        lambda x: " ".join(word.capitalize() for word in x.split())
+        if isinstance(x, str)
+        else x
+    )
+
+    pathology_df["sex"] = pathology_df["sex"].str.capitalize()
+    pathology_df["tissue"] = pathology_df["tissue"].apply(
+        lambda x: " ".join(word.capitalize() for word in x.split())
+        if isinstance(x, str)
+        else x
+    )
+
+    # Replace 'beta' with '&beta;' in biomarker types
+    biomarkers_df["type"] = biomarkers_df["type"].str.replace("beta", "&beta;")
+    pathology_df["type"] = pathology_df["type"].str.replace("beta", "&beta;")
+
+    # Rename 'type' column to 'evidence_type' and 'measurement' to 'value'
+    biomarkers_df = biomarkers_df.rename(
+        columns={"type": "evidence_type", "measurement": "value"}
+    )
+    pathology_df = pathology_df.rename(
+        columns={"type": "evidence_type", "measurement": "value"}
+    )
+
+    # Convert matching controls and aliases from comma-delimited strings to lists
+    model_info_df["matched_controls"] = model_info_df["matched_controls"].apply(
+        lambda x: [item.strip() for item in str(x).split(",")]
+        if (pd.notna(x) and x != "")
+        else []
+    )
+
+    model_info_df["aliases"] = model_info_df["aliases"].apply(
+        lambda x: [item.strip() for item in str(x).split(",")]
+        if (pd.notna(x) and x != "")
+        else []
+    )
+
+    # Build the final data structure
+    result = []
+
+    # Process each model
+    for _, model_row in model_info_df.iterrows():
+        model_name = model_row["model"]
+
+        # Get genetic info for this model
+        genetic_info = []
+        model_alleles = allele_info_df[allele_info_df["model"] == model_name]
+
+        for _, allele_row in model_alleles.iterrows():
+            gene_info = {
+                "modified_gene": allele_row["gene"],
+                "ensembl_id": allele_row["gene_ensembl_id"],
+                "allele": allele_row["allele"],
+                "allele_type": allele_row["allele_type"],
+                "mgi_allele_id": allele_row["mgi_allele_id"],
+            }
+
+            # If it's a human transgene, replace the ensembl_id with the human one
+            if (
+                allele_row["mgi_allele_id"]
```

Comment:
There _might_ be a way to simplify this code a little bit by doing a full join (or left join?) of `allele_info_df` and `human_transgene_allele_map_df` on `mgi_allele_id` outside this for loop, pulling `allele_row` from that merged df, and checking if the gene symbol is NA or not. But without playing around I can't say for sure if that will work.